_posts/2021-01-12-github-application-running-on-faasd-behind-inlets.md

```diff
@@ -0,0 +1,508 @@
+
+In this guide, we are going to develop a [Github application](https://docs.github.com/en/free-pro-team@latest/developers/apps) using Go, then we deploy it as a serverless function to make use of [faasd](https://github.com/openfaas/faasd) which is a lightweight & portable faas engine.We are also going to do this demo on our local environment, so we should open our function which runs on our local environment to the Internet so Github can send events to our function.In order to do that we use inlets-pro which provides secure TCP/L4 tunnels.
+
+## Pre-requisites
+
+- arkade : You can basically use arkade to install of these following tools, so we need to install arkade first.
+arkade provides a portable marketplace for downloading your favourite devops CLIs and installing helm charts, with a single command.
+
+```bash
+$ curl -sLS https://dl.get-arkade.dev | sh
+$ arkade version
+            _             _
+  __ _ _ __| | ____ _  __| | ___
+ / _` | '__| |/ / _` |/ _` |/ _ \
+| (_| | |  |   < (_| | (_| |  __/
+ \__,_|_|  |_|\_\__,_|\__,_|\___|
+
+Get Kubernetes apps the easy way
+
+Version: 0.6.35
+Git Commit: df53c4f6d9c604186b36aae7f0feb1d39940be8f
+```
+
+- inletsctl : We need inletsctl to automate the task of creating an exit-node on cloud infrastructure.
+
+```bash
+$ arkade get inletsctl
+$ inletsctl version
+ _       _      _            _   _
+(_)_ __ | | ___| |_ ___  ___| |_| |
+| | '_ \| |/ _ \ __/ __|/ __| __| |
+| | | | | |  __/ |_\__ \ (__| |_| |
+|_|_| |_|_|\___|\__|___/\___|\__|_|
+
+Version: 0.7.1
+Git Commit: 8641a9a09903f0e3b18a4ed1d388a35663a52dc4
+```
+
+- inlets-pro : We need inlets-pro to tunnel out any TCP traffic from an internal network to another network.
+
+
+```bash
+$ inletsctl download --pro
+$ inlets-pro version
+ _       _      _            _
+(_)_ __ | | ___| |_ ___   __| | _____   __
+| | '_ \| |/ _ \ __/ __| / _` |/ _ \ \ / /
+| | | | | |  __/ |_\__ \| (_| |  __/\ V /
+|_|_| |_|_|\___|\__|___(_)__,_|\___| \_/
+
+PRO edition
+
+Version: 0.7.2
+Git Commit: 5190eca94b4020ec8de47a22eb411bb17a995399
+```
+
+- multipass : We need to provision ubuntu vm's to host our faasd.In order to do that multipass cli is the best tool.
+
+```bash
+$ brew install multipass
+$ multipass version
+multipass  1.5.0+mac
+multipassd 1.5.0+mac
+```
+
+- faas-cli
+We need faas-cli to interact with the gateway which runs on faasd.
+
+```bash
+$ arkade get faas-cli
+$ faas-cli version
+  ___                   _____           ____
+ / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
+| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
+| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
+ \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
+      |_|
+
+CLI:
+ commit:  598336a0cad38a79d5466e6a3a9aebab4fc61ba9
+ version: 0.12.21
+```
+
+Now we are ready to go 🚀
+
+## Setup exit-node server on AWS
+In order to do that, we need two things: AWS Account and inletsctl. <br/>First we need to get our access key ID and the secret access key from our AWS Account. 
```

Comment:
AWS is a partciularly complicated backend to provision to and these instructions are already present in the inlets docs. Instead please use DigitalOcean and remove the AWS instructions.

Link to the docs and say other clouds exist, if they want to use them like AWS or Hetzner.

Reply:
done